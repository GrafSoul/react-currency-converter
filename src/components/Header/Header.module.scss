.header {
  background-color: #004d40;
  color: #fff;
  height: 3rem;
  min-width: 320px;

  .headerContent {
    display: flex;
    justify-content: space-between;
    max-width: 70rem;
    margin: auto;
  }

  .headerName {
    margin-top: 0.3em;
    margin-left: 0.3em;
    font-size: 1.2em;
    font-weight: bold;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .logo {
    display: inline-block;
    margin: 0 0.3rem 0 0;
  }

  .name {
    display: inline-block;
    margin-top: -0.4rem;
  }

  .menu {
    display: flex;
  }

  .menuItem {
    a {
      display: block;
      padding: 0.8rem 1rem;
      height: 3rem;
      color: #fff;

      &:hover {
        background-color: #00836d;
      }
    }
  }

  .active {
    background-color: #01816c;
  }
}

@media only screen and (max-width: 420px) {
  .header {
    height: auto;

    .headerContent {
      justify-content: flex-start;
      flex-direction: column;
    }

    .menu {
      margin-top: 0.2rem;
      border-top: 1px solid #00836d;
      flex-direction: column;
    }

    .menuItem {
      a {
        padding: 0.3rem 1rem;
        height: 2.2rem;
      }
    }
  }
}
