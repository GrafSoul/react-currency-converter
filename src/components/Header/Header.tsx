import React from "react";
import { Link } from "react-router-dom";

import CustomLink from "./CustomLink/CustomLink";

import classes from "./Header.module.scss";

const Header: React.FC = () => {
  return (
    <header className={classes.header}>
      <div className={classes.headerContent}>
        <div className={classes.headerName}>
          <Link to="/">
            <img className={classes.logo} src="images/logo.png" alt="Лого" />
          </Link>
          <span className={classes.name}>Конвертер валют</span>
        </div>

        <nav>
          <ul className={classes.menu}>
            <li className={classes.menuItem}>
              <CustomLink to="/">Валюты</CustomLink>
            </li>
            <li className={classes.menuItem}>
              <CustomLink to="/converter">Конвертер</CustomLink>
            </li>
          </ul>
        </nav>
      </div>
    </header>
  );
};

export default Header;
