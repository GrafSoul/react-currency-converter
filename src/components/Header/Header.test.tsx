/* eslint-disable testing-library/render-result-naming-convention */
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from './Header';

describe('Header component', () => {
    it('Header render', () => {
        render(
            <MemoryRouter>
                <Header />
            </MemoryRouter>,
        );
        expect(screen.getByText('Конвертер валют')).toBeInTheDocument();
    });

    it('Header snapshot', () => {
        const header = render(
            <MemoryRouter>
                <Header />
            </MemoryRouter>,
        );
        expect(header).toMatchSnapshot();
    });
});
