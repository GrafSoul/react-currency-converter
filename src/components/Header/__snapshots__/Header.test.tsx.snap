// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="headerContent"
        >
          <div
            class="headerName"
          >
            <a
              href="/"
            >
              <img
                alt="Лого"
                class="logo"
                src="images/logo.png"
              />
            </a>
            <span
              class="name"
            >
              Конвертер валют
            </span>
          </div>
          <nav>
            <ul
              class="menu"
            >
              <li
                class="menuItem"
              >
                <a
                  class="active"
                  href="/"
                >
                  Валюты
                </a>
              </li>
              <li
                class="menuItem"
              >
                <a
                  class=""
                  href="/converter"
                >
                  Конвертер
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="headerContent"
      >
        <div
          class="headerName"
        >
          <a
            href="/"
          >
            <img
              alt="Лого"
              class="logo"
              src="images/logo.png"
            />
          </a>
          <span
            class="name"
          >
            Конвертер валют
          </span>
        </div>
        <nav>
          <ul
            class="menu"
          >
            <li
              class="menuItem"
            >
              <a
                class="active"
                href="/"
              >
                Валюты
              </a>
            </li>
            <li
              class="menuItem"
            >
              <a
                class=""
                href="/converter"
              >
                Конвертер
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
